torch/csrc/distributed/rpc/ProcessGroupAgent.cpp

```diff
@@ -138,53 +132,56 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
     future->markCompleted();
   }
 
-  enqueue(SendWork(dstRank, std::move(message)));
+  enqueueSend(RpcWork(dstRank, std::move(message)));
   return future;
 }
 
-void ProcessGroupAgent::enqueue(SendWork work) {
-  std::unique_lock<std::mutex> lock(sendQueueMutex_);
-  sendQueue_.emplace_back(std::move(work));
-  lock.unlock();
-
-  workProduceCV_.notify_one();
+void ProcessGroupAgent::enqueueSend(RpcWork work) {
+  threadPool_.run(std::bind(
```

Comment:
Nice trick, `std::bind` to force a move capture.

Could you add a note here that this can be changed to use a native move capture when we move to C++14?